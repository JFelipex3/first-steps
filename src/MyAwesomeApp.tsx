// Tarea
// Crear un componente de React dentro del src lladamo MyAwesomeApp
// El componente debe de retornar (cambiar el nombre)
// <h1>Fernando</h1>
// <h3>Herrera</h3>
// Hacer las configuraciones respectivas para poder ver el componente en pantalla.

import { CSSProperties } from "react";
import { test } from 'vitest';

// export function MyAwesomeApp() {
//     return (
//         <>
//             <h1>Jhonnatan</h1>
//             <h3>Machuca</h3>
//         </>
//     );
// }

// Se pueden definir las variables acá, si no se van a modificar
const firstName = 'Jhonnatan';
const lastName = 'Machuca';

const favoriteGames = ['Elden Ring', 'Smash', 'Metal Gear'];
const isActive = true;

const address = {
    zipCode: '255555040',
    country: 'Chile'
}

const myStyles: CSSProperties = {
    backgroundColor: 'red',
    borderRadius: 10,
    padding: 10,
    marginTop: 30
}

// Version como función de flecha
export const MyAwesomeApp = () => {

    // Definir variables que pueden cambiar acá
    let valor = 'Definido dentro de la función';

    valor = 'Valor actualizado dentro de la función';

    return (
        // los data-testid="div-app" no son recomendados porque pueden ser borados y afectarian al testing
        <div data-testid="div-app">
            <h1 data-testid="first-name-title">{firstName}</h1>
            <h3>{lastName}</h3>

            <p className="mi-clase-favorita">{favoriteGames.join(', ')}</p>
            <p>{2 + 2}</p>

            <h1>{isActive ? 'Activo' : 'No Activo'}</h1>

            {/* Se define doble llave para objetos literales */}
            {/* <p style={{
                backgroundColor: 'red',
                borderRadius: 10,
                padding: 10
            }}> */}
            <p style={myStyles}>
                {JSON.stringify(address)}
            </p>

            <p>{valor}</p>
        </div>
    )
}