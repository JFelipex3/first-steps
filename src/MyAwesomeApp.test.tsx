import { describe, expect, test } from "vitest";
import { render, screen } from '@testing-library/react';

import { MyAwesomeApp } from "./MyAwesomeApp";

describe('MyAwesomeApp', () => {

    test('should render firstname and lastname (container)', () => {

        // Usando container para evaluar estados iniciales - render inicial
        const { container } = render(<MyAwesomeApp />);

        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');

        expect(h1?.innerHTML).toContain('Jhonnatan');
        expect(h3?.innerHTML).toContain('Machuca');

    });

    test('should render firstname and lastname (screen)', () => {

        render(<MyAwesomeApp />); r

        // const h1 = screen.getByRole('heading', {
        //     level: 1
        // });

        // console.log(h1.innerHTML);

        const h1 = screen.getByTestId('first-name-title');
        expect(h1.innerHTML).toContain('Jhonnatan');

    });

    test('should match snapshot (container)', () => {

        // Se usa para evaluar la estructura, si cambiara se daria un error, pero puede ser actualizado el snapshot
        const { container } = render(<MyAwesomeApp />);
        expect(container).toMatchSnapshot();

    });

    test('should match snapshot (screen)', () => {

        render(<MyAwesomeApp />);
        expect(screen.getByTestId('div-app')).toMatchSnapshot();

    });
});