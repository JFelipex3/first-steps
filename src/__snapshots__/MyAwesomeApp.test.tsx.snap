// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapchot (container) 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Jhonnatan
    </h1>
    <h3>
      Machuca
    </h3>
    <p
      class="mi-clase-favorita"
    >
      Elden Ring, Smash, Metal Gear
    </p>
    <p>
      4
    </p>
    <h1>
      Activo
    </h1>
    <p
      style="background-color: red; border-radius: 10px; padding: 10px; margin-top: 30px;"
    >
      {"zipCode":"255555040","country":"Chile"}
    </p>
    <p>
      Valor actualizado dentro de la función
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapchot (screen) 1`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Jhonnatan
  </h1>
  <h3>
    Machuca
  </h3>
  <p
    class="mi-clase-favorita"
  >
    Elden Ring, Smash, Metal Gear
  </p>
  <p>
    4
  </p>
  <h1>
    Activo
  </h1>
  <p
    style="background-color: red; border-radius: 10px; padding: 10px; margin-top: 30px;"
  >
    {"zipCode":"255555040","country":"Chile"}
  </p>
  <p>
    Valor actualizado dentro de la función
  </p>
</div>
`;
